apis/kubedb/v1alpha2/zookeeper_helpers.go

```diff
@@ -193,23 +194,49 @@ func (z *ZooKeeper) SetDefaults() {
 
 	z.setDefaultContainerSecurityContext(&zkVersion, &z.Spec.PodTemplate)
 
+	dbContainer := coreutil.GetContainerByName(z.Spec.PodTemplate.Spec.Containers, ZooKeeperContainerName)
+	if dbContainer != nil && (dbContainer.Resources.Requests == nil && dbContainer.Resources.Limits == nil) {
+		apis.SetDefaultResourceLimits(&dbContainer.Resources, DefaultResources)
+	}
+
 	z.SetHealthCheckerDefaults()
 }
 
 func (z *ZooKeeper) setDefaultContainerSecurityContext(zkVersion *catalog.ZooKeeperVersion, podTemplate *ofst.PodTemplateSpec) {
 	if podTemplate == nil {
 		return
 	}
-	if podTemplate.Spec.ContainerSecurityContext == nil {
-		podTemplate.Spec.ContainerSecurityContext = &core.SecurityContext{}
-	}
 	if podTemplate.Spec.SecurityContext == nil {
 		podTemplate.Spec.SecurityContext = &core.PodSecurityContext{}
 	}
 	if podTemplate.Spec.SecurityContext.FSGroup == nil {
 		podTemplate.Spec.SecurityContext.FSGroup = zkVersion.Spec.SecurityContext.RunAsUser
 	}
-	z.assignDefaultContainerSecurityContext(zkVersion, podTemplate.Spec.ContainerSecurityContext)
+
+	container := coreutil.GetContainerByName(podTemplate.Spec.Containers, ZooKeeperContainerName)
+	if container == nil {
+		container = &core.Container{
+			Name: ZooKeeperContainerName,
+		}
+		podTemplate.Spec.Containers = append(podTemplate.Spec.Containers, *container)
+	}
+	if container.SecurityContext == nil {
+		container.SecurityContext = &core.SecurityContext{}
+	}
+
+	z.assignDefaultContainerSecurityContext(zkVersion, container.SecurityContext)
```

Comment:
I see a serious issue here.  Dereferenced value(`*container`) has been already appended in line 221.  Then we are changing that pointer variable in 227.  Obviously, The above append will not get the securityContext set.

Reply:
As we are passing reference in the assingDefaultContainer function, it will set the securityContext.

Reply:
Nope.  SecurityContext will not be set in the first iteration, as per my understanding.  Can you check `podTemplate.Spec.Containers` value  right after this assignment https://github.com/kubedb/apimachinery/blob/monitor/apis/kubedb/v1alpha2/zookeeper_helpers.go#L221 ?

Reply:
Got the problem. It was assigning in the second call. Thanks